### PR TITLE
Add stub definitions of inject debug resources targets to fix runtimeinfo build dependency issues.

### LINKED ISSUE #51248: Linux build: The dependency target "inject_debug_resources_coreclr" of target "runtimeinfo" does not exist.
Sometime recently, I've become unable to build dotnet/runtime on Linux Ubuntu 20.04.2 LTS.

Steps:

1. `git pull` (HEAD is now b80ddefa8db351f4ed697a686d8b6ba1dd927c52)
2. `git clean -xdf` (just for good measure)
3. `./build.sh -s clr+libs -rc release`

Result:

<pre>
  -- Configuring done
  CMake Error at debug/runtimeinfo/CMakeLists.txt:34 (add_dependencies):
    The dependency target "inject_debug_resources_coreclr" of target
    "runtimeinfo" does not exist.
  
  
  -- Generating done
  CMake Generate step failed.  Build files cannot be regenerated correctly.
  /code/personal/dotnet/runtime/src/coreclr
</pre>

I don't think anything in particular changed in my environment, and nor do I see any other error higher up in the build logs. I tried a branch that is a few weeks old and was able to build it successfully.

Any assistance getting the build working again would be appreciated.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Should fix https://github.com/dotnet/runtime/issues/51248

@vcsjones can you validate?

### CODE REVIEW

**Comment by vcsjones** [comment]
@jkoritzinsky that appears to have fixed the issue for me. Thank you!

File: src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt [hunk 1]
```diff
@@ -243,6 +243,9 @@ if(CLR_CMAKE_TARGET_WIN32)
         add_custom_target(inject_debug_resources_singlefilehost ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/inject_debug_resources_singlefilehost.timestamp)
         add_dependencies(runtime inject_debug_resources_singlefilehost)
     endif()
+else()
+    add_custom_target(inject_debug_resources_coreclr DEPENDS coreclr)
```
Comment (hoyosjs) [inline src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt:247]: The other thing that we could do is make runtimeinfo depend on inject_debug_resources_coreclr on windows, given that conceptually there's no external resources needed for debugging coreclr. Do you have a preference between the dependencies vs this phony target?

Comment (jkoritzinsky) [reply src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt:247]: I'd rather do the phony target so we don't need to add a different dependency that has the same effect on non-Windows platforms.

**Review by hoyosjs — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Failure looks like a machine disconnect. Merging.

**Pull request merged**